### PR TITLE
DM with '@' sign feature (tests needed)

### PR DESCRIPTION
**Pull request by Garanas**
Replaces https://github.com/FAForever/fa/pull/3568 .

Proof of concept fro pm'ing players with @ sign.
Still need to add toggle to enable/disable it + better design for popup.

### CODE REVIEW

**Comment by Garanas** [comment]
@4z0t I think this feature is great - I'll add in a toggle in the chat menu. You mention it still needs tests - what type of tests? One issue is that you can whisper to yourself at the moment 😄

**Comment by 4z0t** [comment]
Afair if you type a text and then do @ thing it clears edit, I wanted to make it only work if edit is empty i.e. @ is first in string

File: lua/ui/game/chat.lua [hunk 1]
```diff
@@ -560,6 +560,86 @@ function FindClients(id)
     return result
 end
 
+function MatchPlayersNicknames(s)
```
Comment (Askaholic) [inline lua/ui/game/chat.lua:563]: Both of these could probably be tested in the unit tests without much trouble. Also why is MatchStrings an inner function? Would it make sense to move it to the string utils file?

Comment (4z0t) [reply lua/ui/game/chat.lua:563]: But it does very specific function here, i dont think it`s needed there

**Comment by Garanas** [comment]
@4z0t I think this has become obsolete by the feature that allows you to pm someone when clicking their name. If you agree then you can close this pull request.

**Comment by 4z0t** [comment]
it is applied in-game chat not lobby one, if i have some time, i may complete it before new patch

**Comment by Garanas** [comment]
I'd like it to be completed yesterday if you want it to be part of the next patch. It needs live testing to ensure the game is stable.

**Comment by 4z0t** [comment]
cant push for some reason...
```! [rejected]            improve-whisper -> improve-whisper (non-fast-forward)
error: failed to push some refs to 'https://github.com/FAForever/fa.git'```

**Comment by 4z0t** [comment]
*kill me*

**Comment by 4z0t** [comment]
thats what happens when old PR get touched xd

**Comment by Garanas** [comment]
Perhaps we should just leave it then - the patch is full of improvements already.

**Comment by Askaholic** [comment]
```
git checkout improve-whisper
git reset --hard 49f33ff
git rebase deploy/fafdevelop
git push --force
```

That should fix it, resetting to the commit `49f33ff` which looks like the last relevant one made before the merge

**Comment by Garanas** [comment]
Closing due to aging

**Pull request closed without merging**